java/google/registry/whois/DomainLookupCommand.java

```diff
@@ -31,7 +33,8 @@ public DomainLookupCommand(InternetDomainName domainName, @Nullable InternetDoma
   }
 
   @Override
-  WhoisResponse getSuccessResponse(DomainResource domain, DateTime now) {
-    return new DomainWhoisResponse(domain, now);
+  protected WhoisResponse getSuccessResponse(InternetDomainName domainName, DateTime now) {
+    final DomainResource domainResource = loadByForeignKey(DomainResource.class, domainName.toString(), now);
+    return domainResource == null ? null : new DomainWhoisResponse(domainResource, now);
```

Comment:
Change the signature to return Optional<WhoisResponse>.  It's weird to just return null sometimes when someone looking at the calling code is likely always expecting a response to be returned.